gdraw/ggdkdraw.c

```diff
@@ -593,23 +608,30 @@ static GWindow _GGDKDraw_CreateWindow(GGDKDisplay *gdisp, GGDKWindow gw, GRect *
             nw->is_dlg = true;
         } else if (!nw->is_dlg) {
             ++gdisp->top_window_count;
-        } else if (nw->restrict_input_to_me) {
-            if (gdisp->last_nontransient_window != NULL) {
-                GGDKDrawSetTransientFor((GWindow)nw, (GWindow) - 1);
-            } else if (gdisp->top_window_count > 0) {
-                Log(LOGWARN, "Could not call GGDKDrawSetTransientFor on a restricted window (%p %s), restrict count %d, transient window count %d",
-                    nw, nw->window_title, gdisp->restrict_count, (int)gdisp->transient_stack->len);
-            }
+        } else if (nw->restrict_input_to_me && gdisp->toplevel_windows->length > 0) {
+            GGDKDrawSetTransientFor((GWindow)nw, (GWindow) - 1);
         }
         nw->isverytransient = (wattrs->mask & wam_verytransient) ? 1 : 0;
-        nw->is_toplevel = true;
+
```

Comment:
This use of `g_list_append` might be worth a comment something like "Create a list link element pointing to the window"